### PR TITLE
unpin torch to use 1.12.1

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

unpin torch to use 1.12.1

### CODE REVIEW

**Comment by LysandreJik** [comment]
Let's maybe add a `!=1.12.0` as well. cc @sgugger, who also has a PR open here: https://github.com/huggingface/transformers/pull/17925

**Comment by ydshieh** [comment]
Here are the tests to fix before we can use 1.12.1

  - tests/models/tapas/test_modeling_tf_tapas.py -k "TFTapasModelTest and test_pt_tf_model_equivalence"
  - tests/onnx/test_onnx_v2.py -k "StableDropoutTestCase and test_training"
  - tests/pipelines/test_pipelines_table_question_answering.py -k "TQAPipelineTests and test_slow_tokenizer_sqa_pt"
  - tests/pipelines/test_pipelines_table_question_answering.py -k "TQAPipelineTests and test_small_model_pt"
  - tests/models/tapas/test_modeling_tapas.py::TapasUtilitiesTest::

**Comment by LysandreJik** [comment]
Most tapas tests will likely work given @sgugger's PR above, as it's probably linked to the version of the torch-scatter dependency

**Comment by ydshieh** [comment]
Yes! Thanks, @LysandreJik

**Comment by ydshieh** [comment]
(guess I can close this PR, and just merge #17925)
I will check `StableDropoutTestCase and test_training` though.

File: src/transformers/models/deberta/modeling_deberta.py [hunk 1]
```diff
@@ -187,6 +187,8 @@ def backward(ctx, grad_output):
 
     @staticmethod
     def symbolic(g: torch._C.Graph, input: torch._C.Value, local_ctx: Union[float, DropoutContext]) -> torch._C.Value:
+        from torch.onnx import symbolic_opset12
```
Comment (ydshieh) [inline src/transformers/models/deberta/modeling_deberta.py:190]: Without this line, we get (for torch `1.12.x`):
```python
FAILED tests/onnx/test_onnx_v2.py::StableDropoutTestCase::test_training - AttributeError: module 'torch.onnx' has no attribute 'symbolic_opset12'
```

File: src/transformers/models/deberta_v2/modeling_deberta_v2.py [hunk 2]
```diff
@@ -193,6 +193,8 @@ def backward(ctx, grad_output):
 
     @staticmethod
     def symbolic(g: torch._C.Graph, input: torch._C.Value, local_ctx: Union[float, DropoutContext]) -> torch._C.Value:
+        from torch.onnx import symbolic_opset12
```
Comment (ydshieh) [inline src/transformers/models/deberta_v2/modeling_deberta_v2.py:196]: Just applied the same change

File: src/transformers/models/sew_d/modeling_sew_d.py [hunk 3]
```diff
@@ -597,6 +597,8 @@ def backward(ctx, grad_output):
 
     @staticmethod
     def symbolic(g: torch._C.Graph, input: torch._C.Value, local_ctx: Union[float, DropoutContext]) -> torch._C.Value:
+        from torch.onnx import symbolic_opset12
```
Comment (ydshieh) [inline src/transformers/models/sew_d/modeling_sew_d.py:600]: same

**Comment by sgugger** [comment]
Can you push necessary changes directly on #17925 (I'm too lazy to check this PR contains the same fixes as this one 😅 ) The branch is `enable_pt12`.

**Comment by ydshieh** [comment]
Close this and work on  #17925 instead

**Pull request closed without merging**